File: server/common/packets/agency_close.py
AGENCY_CLOSE_PACKET_TYPE = "AgencyClose"


class AgencyClose:
    """
    Parse the agency number from the packet bytes received from agency.
    """
    @staticmethod
    def parse_agency(data: bytes) -> str:
        """
        Parse agency name from packet bytes received from agency.
        It should only have the agency number, encoded in UTF-8.
        """

        agency = data.decode('utf-8')

        return agency

File: client/common/packets/agency_close.go
package packets

const AgencyClosePacketType string = "AgencyClose"

type AgencyClosePacket struct {
	Agency string
}

func NewAgencyClosePacket(agency string) *AgencyClosePacket {
	return &AgencyClosePacket{
		Agency: agency,
	}
}

func (packet *AgencyClosePacket) Encode() ([]byte, error) {
	bytes := make([]byte, 0)
	bytes = append(bytes, AgencyClosePacketType...)
	bytes = append(bytes, ':')
	bytes = append(bytes, packet.Agency...)

	return AppendLengthToPacket(bytes)
}
